mantid/wfm/wfm_stitching.md

```diff
@@ -0,0 +1,169 @@
+
+In principle, stitching is only required to be run once, as very little tweaking of the stitching parameters should be necessary. In addition, some users were carrying out a study that was comparing WFM and single-pulse data using the same sample set-up, and were having to maintain two different (but very similar) reduction scripts, depending on whether they were reducing WFM or single-pulse data.
+
+We thus propose that WFM stitching could be implemented as a post-processing step in the data acquisition. The raw data could be read in, stitched using the parameters from the beamline (the Nexus geometry is contained inside the data file), and a new post-processed file would then be written to disk (and added to SciCat). All the parameters that were used for the stitching would be included in the file metadata. This would effectively alleviate all the issues listed above.
+
+The users would still have access to the raw un-stitched data, so that they are able to perform the stitching themselves, if they suspect something went wrong during that phase. But for 99% of cases, assuming auto-reduction is operating to satisfaction at an ESS facilty in full production mode, automatic post-processed stitching should work nicely.
+
+Going one step further, one could even imagine including the event-by-event stitching (although it makes less sense to call it 'stitching' here, 'conversion to time-of-flight' may be more appropriate) as part of the filewriter. It should be possible to write the raw and stitched data files simultaneously, thus cutting out the time needed to re-read the raw data to perform the conversion.
+
+## 5. Conclusion and distribution
+
+We conclude that both peak-finding and analytical predictions for WFM frame boundary determination should be complementary. While Method 1 presents several major drawbacks that make it unfit to a production set-up, it should prove extremely useful as a diagnostic or verification tool during the hot commissioning phase.
+
+Currently, the different methods are scattered across different repositories, on both Gitlab and Github, with virtually no documentation. Moving forward, we should ensure that everything is put in one place, alongside appropriate documentation.
+
+The programs to perform the WFM stitching should be distributed as follows:
+
+1. A separate package containing the frame boundary determination, which would be distributed on `pip` and `conda`. This should aim to remain Python-only, with no large dependencies such as Mantid. This package would allow a user to decide which method they want to use to obtain the frame boundaries: either using the peaking finding, or the TOF diagram, or just a manual input. This package would also provide a basic (non-optimized) method to perform the TOF conversion, that could be used by e.g. Mantid users who do not wish to use the Scipp package.
```

Comment:
Are we going to make this independent of the loading. The present WFM package loads the frames using the fits format (we had problems with this as you know). Do we exclude the loading from the WFM?

Reply:
I think it should be independent of loading if possible, since as you say there is fits files loading, and also loading with Mantid for nexus files.
I guess in Python it should be easy to make it so that it can accept different kinds of inputs. It could accept a Mantid workspace, or some raw count data. Not sure how complicated/maintanable that would get...